Feature/53 test uses proof property in did document

ITStar10 opened the pull request.

ITStar10 committed:
Working...

Commit:
Get proof test working on the javascript side.

ITStar10 committed:
VDA-Verification-Base : With sub feature tests included

Should be tagged for later use

ITStar10 committed:
VDA-Verification-Base : 0.0.2

- Updated the proof string generation.
	Not used ethers.utils.solidityPack() function. It's a raw string of 2 addresses now
- Created test that uses proof property in DID document
	Read proof from the DIDDocument and used that to verify request

tahpot commented on VDA-Verification-Base/test/proof.ts:
```diff
@@ -0,0 +1,184 @@
+            )
+            const signature = await keyring.sign(params)
+
+            await contract.verifyRequestWithArray(ADDRESS, [ADDRESS], params, signature, didDocumentContextProof);
```
Should get the response and assert it is okay

ITStar10 replied:
See line number 180. It checks whether the nonce updated.

tahpot commented on VDA-Verification-Base/test/proof.ts:
```diff
@@ -0,0 +1,184 @@
+        await contract.deployed()
+    })
+
+    describe('Keyring test', () => {
```
Can this test be removed now?

ITStar10 replied:
I think, we can leave it for later reference.

tahpot reviewed (commented):
What if we merged into a single function that uses the provided `validSigners` if provided, otherwise it uses the signers stored in the contract?

```
EnumerableSetUpgradeable.AddressSet protected _trustedSigners

function verifyRequest(
        address did, 
        bytes memory params, 
        bytes calldata signature, 
        bytes calldata proof,
        address[] calldata validSigners,
    ) {
  
  index = 0
  length = 0
  memorySigner = false
  if (validSigners.length == 0) {
    memorySigner = true
    length = this._trustedSigners.length()
  } else {
    memorySigner = false
    length = validSigners.length
  }

  while (index < length && !isVerified) {
    if (memorySigner) {
        account = this._trustedSigners.at(index)
    } else {
        account = validSigners[index]
    }

    ... rest of the code
  }
```

I think it makes sense that this contract specifies the variable name for the white list of trusted signers (`_trustedSigners`). It should also support adding and removing them, so you could move that from the DID linkage branch.

Thoughts?

ITStar10 committed:
V 0.0.2 - Minor updates on review

- Added comments to the functions
- Move the verifyRequestWithArray(...) function to the TestContract as it's not used in the practice.

ITStar10 committed:
V 0.0.2 - Add trusted signers to the VDA-Verification-Base contract

Updated
	- Add `_trustedSigners` variable
	- `addTrustedSigner()` & `removeTrustedSigner()` functions added
	- Overload the `verifyRequest()` function

Commit:
Update proof unit test to better demonstrate what the test needs to achieve.

Commit:
Add placeholder examples

Commit:
Add nonce to signed request params

Commit:
Refactor to support proof capabilities

Commit:
Merge branch 'feature/53-test-uses-proof-property-in-DID-document' of https://github.com/verida/blockchain-contracts into feature/53-test-uses-proof-property-in-DID-document

Commit:
Delete accidental commits

ITStar10 committed:
Updates : KeyringTest() in proof.ts is working

ITStar10 committed:
Update : All test working

Commit:
Remove deprecated dependencies and some other code cleanup
